oio/common/cryptography_tools.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (C) 2016 OpenIO SAS
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+from cryptography.fernet import Fernet
+from cryptography.fernet import InvalidToken
+from oio.common.exceptions import SourceReadError, OioException
+
+
+class CryptographyToolsException(Exception):
+    def __init__(self, message):
+        super(CryptographyToolsException, self).__init__()
+        self._message = message
+
+    def __str__(self):
+        return self._message
+
+
+# This cryptography tool use the Fernet method of the cryptography module
+# to work. If this cryptography tool has vuln, change this tool to use the new
+# cryptography module.
+# the from_crypt_function_to_actual_crypt take into account this possibility
+# and can change the encryption of the data
+class CryptographyTools(object):
+    READ_SIZE = 65536
+    BLOCK_SIZE = 128
+    VERSION_SIZE = 8
+    TIMESTAMP_SIZE = 64
+    HMAC_SIZE = 256
+    crypt = None
+
+    def __init__(self, key=None):
+        if key:
+            self.crypt = Fernet(key)
+
+    @staticmethod
+    def generate_key():
+        return Fernet.generate_key()
+
+    def encrypt(self, message):
+        if not self.crypt:
+            return message
+        try:
+            return self.crypt.encrypt(message)
+        except TypeError:
+            raise CryptographyToolsException("Error cryptography")
+
+    def decrypt(self, cipher):
+        if not self.crypt:
+            return cipher
+        try:
+            return self.crypt.decrypt(cipher)
+        except TypeError:
+            raise CryptographyToolsException('Error cryptography')
+        except InvalidToken:
+            raise CryptographyToolsException('Invalid Token')
+
+    # if the precedent cryptography system has been compromised
+    # this function can be used to passing to the new cryptography system
+    def from_crypt_function_to_actual_crypt(self, decrypt, cipher):
+        decrypt_content = decrypt(cipher)
+        return self.encrypt(decrypt_content)
+
+    def _generate_dict(self, over, bytes_read, ciphered_bytes, content=None):
+        """
+        generate a dictionnary with some informations,
+        see the function read_and encrypt for more details
+        """
+        return {'over': over,
+                'bytes_read': bytes_read,
+                'ciphered_bytes': ciphered_bytes,
+                'content': content}
```

Comment:
To generate a dictionary, you call `dict()`. The purpose of this function is not to _generate a dictionary_, so please rename it.